viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
 # ------------- WAVE 1 --------------------
 
+from enum import unique
+
+# This is a helper function used in some above functions
+def get_users_watched_movie_titles(user_data):
+    users_watched_movies = []
+    for film in user_data["watched"]:
+        users_watched_movies.append(film["title"])
+    return users_watched_movies
+
 def create_movie(title, genre, rating):
-    pass
+    movie = {"title" : title, "genre" : genre, "rating" : rating}
+    
+    if not title or not genre or not rating: 
+        movie = None
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+    
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    movie_counter = 0
+    rating_total = 0
+
+    for movie in user_data["watched"]:
+        movie_counter += 1
+        rating_total += movie["rating"]
+    if movie_counter == 0:
+        average_rating = 0.0
+    else: 
+        average_rating = rating_total / movie_counter
+    
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    
+    if not user_data["watched"]:
+        return None
+    else: 
+        genre_dict = {}
+        for movie in user_data["watched"]:
+            if movie["genre"] in genre_dict:
+                genre_dict[movie["genre"]] += 1
+            else:
+                genre_dict[movie["genre"]] = 1
+        
+        popular_genre = max(genre_dict, key=genre_dict.get)

```

Comment:
Nice use of `max()` to find the genre with the highest count. How would you do something similar if a function like `max()` didn't exist? Now is a great time to think through how library functions solve these problems!